core/src/main/java/org/apache/gravitino/storage/relational/service/UserMetaService.java

```diff
@@ -265,4 +281,60 @@ public int deleteUserMetasByLegacyTimeline(long legacyTimeline, int limit) {
 
     return userDeletedCount[0] + userRoleRelDeletedCount[0];
   }
+
+  private static class ListSkippingRolesHandler
+      implements SupportsSkippingFields<List<UserEntity>> {
+    private final String metalakeName;
+
+    ListSkippingRolesHandler(String metalakeName) {
+      this.metalakeName = metalakeName;
+    }
+
+    @Override
+    public List<Field> supportsSkippingFields() {
+      return Lists.newArrayList(UserEntity.ROLE_IDS, UserEntity.ROLE_NAMES);
+    }
+
+    @Override
+    public List<UserEntity> execute() {
+      List<UserPO> userPOs =
+          SessionUtils.getWithoutCommit(
+              UserMetaMapper.class, mapper -> mapper.listUserPOsByMetalake(metalakeName));
+      return userPOs.stream()
+          .map(
+              po ->
+                  POConverters.fromUserPO(
+                      po,
+                      Collections.emptyList(),
+                      AuthorizationUtils.ofUserNamespace(metalakeName)))
+          .collect(Collectors.toList());
+    }
+  }
```

Comment:
SkippingFields and execution are just a mapping and are not directly related, e.g. execution needs to use the values in SkippingFields.